### ISSUE TITLE
Type mismatch error reported when using a constrained `TypeVar` to annotate a function when constrained variable is not used in the return expression

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#2428]
**Describe the bug**
Pyright incorrectly reports a type mismatch on the return type when using a constrained `TypeVar` to type annotate a function that does not use the constrained variable to create the returned object through some expression.

Example explains it better:

```py
from typing import TypeVar

T = TypeVar("T", str, int)

def some_func(some_var: T) -> T:
    if isinstance(some_var, str):
        return "some_string" # Type "Literal['some_string']" cannot be assigned to type "T@some_func"
    return 1 # Type "Literal[1]" cannot be assigned to type "T@some_func"
```

here pyright gives an error, even though the type correctly matches the constrained `TypeVar`. Mypy does not report an error in this case.

This however works fine:

```py
from typing import TypeVar

T = TypeVar("T", str, int)

def some_func(some_var: T) -> T:
    if isinstance(some_var, str):
        return some_var + "some_string"
    return some_var + 1
```

It seems like if the constrained variable is not used through some expression to create the returned object, then pyright will report an error.

Tested on pyright 1.1.178

**Comment by erictraut** [comment]
I figured out a way to handle this case. It will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.198, which I just published. It will also be included in the next release of pylance.
